src/wordpress-playground/index.tsx

```diff
@@ -6,10 +6,50 @@ import {
 } from '../php-wasm-browser/index';
 import { ProgressObserver, ProgressType } from './progress-observer';
 import { PromiseQueue } from './promise-queue';
+import { saveAs } from 'file-saver';
+import { DOCROOT } from './config';
+import migration from './migration.php';
 
 const query = new URL(document.location.href).searchParams as any;
 
 const wpFrame = document.querySelector('#wp') as HTMLIFrameElement;
+const addressBar = document.querySelector('#url-bar')! as HTMLInputElement;
+
+let wpVersion;
+let phpVersion;
+
+// Migration Logic
+const importWindow = document.querySelector('#import-window') as HTMLElement;
+const overlay = document.querySelector('#overlay') as HTMLElement;
+const exportButton = document.querySelector(
+	'#export-playground--btn'
+) as HTMLButtonElement;
+const importOpenModalButton = document.querySelector(
+	'#import-open-modal--btn'
+) as HTMLButtonElement;
+const importPlaygroundForm = document.querySelector(
+	'#import-playground-form'
+) as HTMLFormElement;
+const importSelectFile = document.querySelector(
+	'#import-select-file'
+) as HTMLInputElement;
+const importSelectFileText = document.querySelector(
+	'#import-select-file--text'
+) as HTMLElement;
+const importSelectFileButton = document.querySelector(
+	'#import-select-file--btn'
+) as HTMLButtonElement;
+const importSubmitButton = document.querySelector(
+	'#import-submit--btn'
+) as HTMLButtonElement;
+const importCloseModalButton = document.querySelector(
+	'#import-close-modal--btn'
+) as HTMLButtonElement;
```

Comment:
It's about time this entire file gets migrated to (p)react – that's definitely out of scope of this PR, though